### DISCUSSION TITLE
BIGTREETECH Octopus Pro V1.1 STM32H723ZE trying to program

### DISCUSSION THREAD

**Discussion by MeisterPilaf** [stm32duino/Arduino_Core_STM32#2349] [Q&A]
Hello everyone, I am trying to use the BIGTREETECH Octopus Pro V1.1 board for my project. It is my first time using the STM32 MCU (STM32H723ZE) and the STM32duino project, so I don't know what I am doing wrong.I am trying to upload a simple blink sketch via Arduino DFU or compile and put the *.bin file on the SD card to flash the board. Im able by flashing the board using the SD card but no blink. If I do that with the DFU mode, I always overwrite the bootloader which I obtained from the GitHub repository of BTT.

https://github.com/bigtreetech/BIGTREETECH-OCTOPUS-Pro/blob/master/Hardware/BIGTREETECH%20Octopus%20Pro%20V1.1-Pin.jpg

my sketch: 
```c++
#include <Arduino.h>

void setup() {
  // Setze den Pin PA8 als Ausgang
  pinMode(PA_8, OUTPUT);
}

void loop() {
  // Schalte den Pin PA8 ein
  digitalWrite(PA_8, HIGH);
  delay(1000); // Warte eine Sekunde

  // Schalte den Pin PA8 aus
  digitalWrite(PA_8, LOW);
  delay(1000); // Warte eine Sekunde
}
```

My settings on Arduino IDE:
<img width="518" alt="image" src="https://github.com/stm32duino/Arduino_Core_STM32/assets/34895606/68e3f4e0-7398-4c74-b95b-a37787b4a238">

also tried with that:
<img width="518" alt="image" src="https://github.com/stm32duino/Arduino_Core_STM32/assets/34895606/cc3a2598-bea2-4e04-b39b-aad3bff9335b">

Both are not working. I am not able to get the board to trigger the PA_8 pin.

I am pretty sure I have to configure something else in order to program it correctly, but I have no clue what to do.

Maybe someone here can help me.

Thank you in advance

**Comment by fpistm** [comment]
Use `PA8` instead of `PA_8`.
`PYn` are pin number used for the Arduino API, while `PY_n` are `PinName` mainly used internally by the core.

**Reply by MeisterPilaf** [reply]
I have already tried, but there is no difference. It is not working.

This is how I upload the sketch: I create a binary file, copy it to the SD card, rename it to "firmware.bin", insert the SD card into the board, power it on, and the Status LED blinks for a few seconds. I then reconnect the power, but nothing happens.

When I check the SD card, the file is renamed as it should be to "FIRMWARE.CUR".Am I doing something wrong? 

Do I need to configure the "generic board"?

**Reply by fpistm** [reply]
I don't know this board, if you have a bootloader you probably need to build with a VECT_TAB_OFFSET.
The generic is provided as is and use the HSI clock. You can reconfigure the system core clock config if you want but it should work with HSI except if there is a mistake in the default config. :wink: Best is to try to debug.

**Reply by MeisterPilaf** [reply]
How can i change the VECT_TAB_OFFSET? Sorry for the stupid questions but i am a newbie in stm32 world... 

The bootloader comes from BIGTREETECH and it allows me to flash the Board via SD Card. is there a other bootloader i can use instead?

The Bootloader stops at 0x08006A50
[Octopus-H723-bootloader-128KB.hex.zip](https://github.com/stm32duino/Arduino_Core_STM32/files/15095671/Octopus-H723-bootloader-128KB.hex.zip)

**Reply by fpistm** [reply]
See the wiki.

**Reply by MeisterPilaf** [reply]
I successfully made the change in the boards.txt file, but there is no difference. It now starts at 0x6A60 and does not overwrite the bootloader anymore. However, the LED is still not blinking. I don't know what's wrong.

**Reply by fpistm** [reply]
You use PA8? Not PA_8?

**Reply by MeisterPilaf** [reply]
yes:

`#include <Arduino.h>

void setup() {
  // Setze den Pin PA8 als Ausgang
  pinMode(PA8, OUTPUT);
}

void loop() {
  // Schalte den Pin PA8 ein
  digitalWrite(PA8, HIGH);
  delay(1000); // Warte eine Sekunde

  // Schalte den Pin PA8 aus
  digitalWrite(PA8, LOW);
  delay(1000); // Warte eine Sekunde
}`

**Reply by fpistm** [reply]
Looking at the scematics, PA8 is not connected to a LED it is FAN_0 signal connected to a 74HCT365...

PB0 is connected to LED signal, anyway it seems used to drive a RGB LED.

**Reply by MeisterPilaf** [reply]
Yes, but in the schematics, you can see that PA8 is also connected to an LED as an indicator that FAN_0 is ON. Also, I measured with my multimeter if the pin triggers, and I see no 3.3V for 1 second.

**Reply by fpistm** [reply]
Yes but also link to other signal:
![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/20641798/5d40e0bc-e9ea-4264-9db8-206a0a51c5e5)

Anyway if you was able to measure PA8 the code probably stuck before.
As stated try to debug to see if the clock config is properly done as they call error handler in case of error...

**Comment by MeisterPilaf** [comment]
I tried it now with the STM32CubeIDE, and it is working. I can now blink PA8.

` HAL_GPIO_WritePin(GPIOA, GPIO_PIN_8, GPIO_PIN_RESET);
  /* USER CODE END 2 */

  /* Infinite loop */
  /* USER CODE BEGIN WHILE */
  while (1)
  {
    /* USER CODE END WHILE */
	  HAL_GPIO_TogglePin(GPIOA, GPIO_PIN_8);
	      HAL_Delay(1000);
    /* USER CODE BEGIN 3 */
  }
  /* USER CODE END 3 */
}`

I did not configure anything except for PA8 set to GPIO_OUTPUT. I just selected the correct MCU and changed the main.c with the code above.

Why does it not work when I'm using the Arduino IDE?

**Reply by fpistm** [reply]
Probably the clock config... try to comment it to see if it works with default one. Or use the one from your cube ide project.

**Reply by MeisterPilaf** [reply]
where can i find the clock config? Sorry for the questions.

**Comment by fpistm** [comment]
Here:
https://github.com/stm32duino/Arduino_Core_STM32/blob/d5ee71201c219a59c0e4f0ab45261bd4e87fe821/variants/STM32H7xx/H723Z(E-G)T_H730ZBT_H733ZGT/generic_clock.c#L22

We agreed you use the generic and not the nucleo?

**Reply by MeisterPilaf** [reply]
OK I found that file but im not sure what i have to copy from STM32CubeIDE to that file. Could you give me another hint? 🗡️ 

Yes, we are using the generic one.

**Reply by MeisterPilaf** [reply]
[blink.zip](https://github.com/stm32duino/Arduino_Core_STM32/files/15098727/blink.zip)
here is the complete project from STM32CubeIDE. also tried to flash the *.bin file with CubeProgrammer and it is working.

**Comment by MeisterPilaf** [comment]
Ok, i found the "problem":

<img width="807" alt="image" src="https://github.com/stm32duino/Arduino_Core_STM32/assets/34895606/e8d8523b-67ef-4639-8862-eac3e34194db">

If I set this option to "None", it works. Why does that happen?
Do I have to set up the SystemClock? Are there any advantages to using an external oscillator? 

Is the MCU running at 550MHz?

Thank you for your help :)

**Reply by fpistm** [reply]
You have to provide a clock config to ensure the USB clock is 48MHz.
You use STM32CubeMX to properly configure the clock.
About the usage of external oscillator, it allows to have a better stability but it consumes more power.
